clients/java/signalr/src/main/java/com/microsoft/aspnet/signalr/HubConnection.java

```diff
@@ -22,8 +22,13 @@
     private HubConnectionState connectionState = HubConnectionState.DISCONNECTED;
     private Logger logger;
     private List<Consumer<Exception>> onClosedCallbackList;
+    private boolean skipNegotiate = true;
+    private NegotiateResponse negotiateResponse;
+    private String accessToken;
 
-    public HubConnection(String url, Transport transport, Logger logger){
+    private static int MAX_NEGOTIATE_ATTEMPTS = 5;
```

Comment:
Just arbitrarily put 5 for the max. I think it's much higher for TS

Reply:
It's 100 in .NET: https://github.com/aspnet/SignalR/blob/release/2.2/src/Microsoft.AspNetCore.Http.Connections.Client/HttpConnection.cs#L28